docs/flask.rst

```diff
@@ -0,0 +1,326 @@
+
+To override the config value of ``DEBUG`` the environment variable would be
+called ``ACME_DEBUG``.
+
+.. _python-decouple: https://pypi.python.org/pypi/python-decouple
+.. _flask-environ: https://github.com/uniphil/flask-environ
+.. _Flask-Env: https://github.com/brettlangdon/flask-env
+
+.. _flask-serving:
+
+``PORT``
+--------
+
+.. epigraph::
+
+   Listen on environment variable ``$PORT`` for HTTP requests.
+
+Depending on which WSGI server you are using to run your Python application
+there are different ways to accept the :envvar:`PORT` as the port to launch
+your application with.
+
+Gunicorn
+~~~~~~~~
+
+Gunicorn automatically will bind to the hostname:port combination of
+``0.0.0.0:$PORT`` if it find the :envvar:`PORT` environment variable.
+That means running gunicorn is as simple as using this, for example::
+
+    gunicorn <project>:app --workers 4
+
+.. seealso::
+
+    The `full gunicorn documentation <http://docs.gunicorn.org/>`_
+    for more details.
+
+uWSGI
+~~~~~
+
+For uWSGI all you have to do is to bind on the :envvar:`PORT` when you
+define the ``uwsgi.ini``, e.g.:
+
+.. code-block:: ini
+
+    [uwsgi]
+    http-socket = :$(PORT)
+    master = true
+    processes = 4
+    module = <project>:app
+    chdir = /app
+    enable-threads = True
+
+.. seealso::
+
+    The `full uWSGI documentation <http://uwsgi-docs.readthedocs.io/>`_
+    for more details.
+
+.. _flask-versions:
+
+Versions
+--------
+
+.. epigraph::
+
+   Must have a JSON version object at /app/version.json.
+
+Dockerflow requires writing a `version object`_ to the file
+``/app/version.json`` as seen from the docker container to be served under
+the URL path ``/__version__``.
+
+To facilitate this python-dockerflow comes with a Flask view to read the
+file under path the parent directory of the Flask app root. See the
+:class:`Flask API docs <~flask.Flask>` for more information about the
+app root path.
+
+If you'd like to override the location from which the view is reading the
+``version.json`` file from, simply override the optional ``version_path``
+parameter to the :class:`~dockerflow.flask.app.Dockerflow` class, e.g.::
+
+    from flask import Flask
+    from dockerflow.flask import Dockerflow
+
+    app = Flask(__name__)
+    dockerflow = Dockerflow(app, version_path='/app')
+
+Alternatively if you'd like to completely override the way the version
+information is read use the
+:meth:`~dockerflow.flask.app.Dockerflow.version_callback` decorator to
+decorate a callback that gets the ``version_path`` value passed. E.g.::
+
+    import json
+    from flask import Flask
+    from dockerflow.flask import Dockerflow
+
+    app = Flask(__name__)
+    dockerflow = Dockerflow(app)
+
+    @dockerflow.version_callback
+    def my_version(root):
+        return json.loads(os.path.join(root, 'acme_version.json'))
+
+.. _version object: https://github.com/mozilla-services/Dockerflow/blob/master/docs/version_object.md
+
+.. _flask-health:
+
+Health
+------
+
+TODO
```

Comment:
I am not sure how to set up lbhealthchecks?

Reply:
it looks like the healthcheck endpoint runs through configured checks and the lbhealthcheck simply returns 200 if the port is responding?